pkg/frontend/openshiftclustercredentials_post.go

```diff
@@ -60,5 +60,10 @@ func (f *frontend) _postOpenShiftClusterCredentials(ctx context.Context, r *http
 
 	doc.OpenShiftCluster.Properties.ServicePrincipalProfile.ClientSecret = ""
 
-	return json.MarshalIndent(converter.ToExternal(doc.OpenShiftCluster), "", "    ")
+	ext := converter.ToExternal(doc.OpenShiftCluster)
+	if ext == nil {
+		return nil, api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeRequestNotAllowed, "", "Request is not allowed in this API.")
```

Comment:
I wonder if we should reuse `api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeInvalidResourceType, "", "The resource type '%s' could not be found in the namespace '%s' for api version '%s'.", vars["resourceType"], vars["resourceProviderNamespace"], vars["api-version"])` - it's a horrid message but I took it from some other production RP.